src/main/java/appeng/api/implementations/items/IAEWrench.java

```diff
@@ -16,10 +16,19 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 
+import appeng.integration.IntegrationType;
+import appeng.transformer.annotations.Integration.Interface;
+import appeng.transformer.annotations.Integration.InterfaceList;
+import buildcraft.api.tools.IToolWrench;
+import cofh.api.item.IToolHammer;
+
 /**
- * Implemented on AE's wrench(s) as a substitute for if BC's API is not available.
+ * Implemented on AE's wrench(s), automatically pulls in CoFH and BC's interfaces if present
  */
-public interface IAEWrench {
+@InterfaceList(
+        value = { @Interface(iface = "cofh.api.item.IToolHammer", iname = IntegrationType.CoFHWrench),
+                @Interface(iface = "buildcraft.api.tools.IToolWrench", iname = IntegrationType.BuildCraftCore) })
+public interface IAEWrench extends IToolHammer, IToolWrench {
```

Comment:
This is probably not a good way to implement tools from other mods, as
- This is technically a breaking change to API. e.g. OC implements `IAEWrench` but also implements `IToolWrench` and `IToolHammer`, so it's actually not a breaking change for this mod https://github.com/GTNewHorizons/OpenComputers/blob/master/src/main/scala/li/cil/oc/common/item/Wrench.scala#L17 But we can't know what mod outside of NH might be implementing it.
- No other classes in `api` package uses `@Interface`. With this change we enforce mods compiling with AE2 api to have CoFH and BC in dev env if they have reference to `IAEWrench`.

Reply:
I suppose so, was not really thinking too far ahead, but I don't think your points are too worrysome:
1) can use default methods
2) These will be compileOnly references. we can switch to compileOnlyApi and dependent mods should still be fine should they switch to the new api version

Reply:
1. Sure, it'll work
2. Uhh I don't like it, it makes addons to pull unnecessary deps, but I understand it's trade-off with code redundancy. It's ok, BC and CoFH are common mods in 1.7.10 modding scene.